woocommerce/woocommerce-android
# Free Trial: Fix Subscription view UI issues

Summary
==========
Fix issues #8586, #8589, #8587 and add previews for the #8591 scenarios. Since all these issues were minor adjustments, I decided to do them all simultaneously as a single improvement for the Subscription view.

Changelog
==========
- #8586 
Got fixed by adding the vertical scroll Modifier to the View root column so the view doesn't get squeezed with the smaller view height in landscape mode.

- #8587 
Partially fixed through a `pluralizedDays` extension function to the `Period` type. This way, we can generate the correct string translation following localization settings. Tests were added to enforce this behavior too. This improvement will be introduced for the banner in another PR.

- #8589 
Fixed by removing the extra period in the `strings.xml` file. 

- #8591 
Didn't require any fix. The RTL support is working as expected and as mentioned inside the issue comments, the strangeness was probably because of the LTR string. Still, a `@Preview` setting for RTL was added inside the `UpgradesView` so we can adequately verify if any changes we do still keep the expected behavior for RTL mode.

Screenshots
==========
| Before  | After |
| ------------- | ------------- |
| ![Screenshot_20230321_001338](https://user-images.githubusercontent.com/5920403/226509781-b228d73c-d77c-40e3-ab30-c645cc86a355.png) | ![Screenshot_20230320_231411](https://user-images.githubusercontent.com/5920403/226509788-9cfbb792-3e10-4fba-8daa-8f407ed504b6.png) |
| ![Screenshot_20230321_001346](https://user-images.githubusercontent.com/5920403/226509809-bf65a8c7-6183-4bc1-ae06-593331617662.png) | ![Screenshot_20230320_230942](https://user-images.githubusercontent.com/5920403/226509813-2a218c1c-7a64-4bf7-b9a9-eb74f98a408b.png) |

How to Test
==========
1. Configure the app with a Free Trial store plan with just one day left in trial.
2. Open the App and go to the bottom bar menu
3. Once there, hit the `Upgrades` button to enter the Subscription view
4. Verify that the `1 days` info now shows as `1 day` as expected.
5. Verify that the additional period in the `...keep your store running.` string is removed
6. Verify that the UI is correctly presented in landscape mode

Update release notes:
